### ISSUE TITLE
[services] Visualizing more than 10 lines for services logs

### ISSUE THREAD

**Issue by leosala** [cockpit-project/cockpit#8271]
Hallo,

I was having a look at Cockpit for giving some power users access to system journals and services restart over a webinterface, and it seems a really cool tool! On thing which is puzzling me atm is that only the latest 10 lines of the service journal are displayed: is there a way to configure a different default? I had a quick look at the code, and it seems that a `lines` parameter is passed to the `journal` function, but javascript is really not my strongest skill :) Any idea how to do this, either through a configuration parameter or hacking the code?

Thanks!

**Comment by petervo** [comment]
In cockpit we actively try to avoid configuration.

Here's the code that limits it to 10. https://github.com/cockpit-project/cockpit/blob/master/pkg/systemd/init.js#L721

I think there might be some design considerations here though. A better approach might be to keep the limit but have a link to click through to the main log page  (/system/logs) filtered for that service. That of course requires adding some additional filtering functionality to the logs page. 

@andreasn thoughts?

**Comment by andreasn** [comment]
I like the idea of clickthrough to the full journal that will show all the messages for the specific service.
I'll come up with a design for it.

**Comment by andreasn** [comment]
@leosala would that address your current problems?

**Comment by leosala** [comment]
@andreasn it could, thanks a lot! One question from a newbie in Cockpit: is the full journal you are referring to using the full output of `journalctl` (e.g. including also bash output from custom services)? I am asking because I had the impression that some items were filtered out, but haven't had the time for properly testing it.  
Anyhow, I also like the idea of adding services filtering to the main journal page :+1:

**Comment by KKoukiou** [comment]
This is available since https://github.com/cockpit-project/cockpit/releases/190
This was implemented with;
```
commit 2f7d00f248e5e7a9b0b48e1de04792635c43e442
Author: Matej Marusak <mmarusak@redhat.com>
Date:   Sat Mar 16 06:29:55 2019 +0100

    logs: Enabled filtering by service
    
    Enable filtering by all services that in given rules would not return
    empty list.
    
    Closes #8347
```

**Comment by marusak** [comment]
Not really, it was only part of that problem. Now you can filter specific journal entries by service, but there is no link from service details page into journal. This needs to be implemented.
The other direction (from journal to service) is tracked in https://github.com/cockpit-project/cockpit/issues/53

Both should be fairly simple and I believe it is useful, thus reopening.
